src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -98,7 +98,12 @@ public static string GetChecksumsLogInfo(IEnumerable<Checksum> checksums)
         /// This structure stores the 20-byte SHA 1 hash as an inline value rather than requiring the use of
         /// <c>byte[]</c>.
         /// </summary>
-        [StructLayout(LayoutKind.Explicit, Size = 20)]
+        /// <remarks>
+        /// Pack = 4 is specified to work around a Mono runtime behavior where on 64-bit the size of the
+        /// struct would be 24 and not 20 due to alignment (it ignores Size = 20, but Pack = 4 does it).
+        /// Without this fix the ctor would throw because it would use sizeof(Sha1Hash) == 24.
+        /// </remarks>
+        [StructLayout(LayoutKind.Explicit, Size = 20, Pack = 4)]
```

Comment:
Why does this even use StructLayout at all? Are we using this for blitting with something else? Can we update the doc to point to where that is being done?

Reply:
This is a question for @sharwell I presume. I would prefer to avoid digging too deep into this code since I don't understand it.

Reply:
@jasonmalinowski Considering this is a `private` nested type, the current comment and usage ~50 lines above seems to address the situation. It was certainly enough for Kirill to make the correct change in response to the Mono bug (coming in from never seeing it before).

Reply:
My motto is if somebody had to ask, your comments are never sufficient. :smile: Is the blitting above the only reason? If I just rewrote the ToString to not allocate the array entirely...could I remove this?